ambry-network/src/main/java/com.github.ambry.network/SSLTransmission.java

```diff
@@ -201,16 +211,19 @@ private void handshake() throws IOException {
               "SSLHandshake NEED_WRAP channelId {}, appReadBuffer pos {}, netReadBuffer pos {}, netWriteBuffer pos {}",
               getConnectionId(), appReadBuffer.position(), netReadBuffer.position(), netWriteBuffer.position());
           handshakeResult = handshakeWrap(write);
-          if (handshakeResult.getStatus() == SSLEngineResult.Status.BUFFER_OVERFLOW) {
-            int currentPacketBufferSize = packetBufferSize();
-            netWriteBuffer = Utils.ensureCapacity(netWriteBuffer, currentPacketBufferSize);
-            if (netWriteBuffer.position() >= currentPacketBufferSize) {
-              throw new IllegalStateException("Buffer overflow when available data size (" + netWriteBuffer.position()
-                  + ") >= network buffer size (" + currentPacketBufferSize + ")");
+          if (handshakeResult.getStatus() == Status.BUFFER_OVERFLOW) {
+            int currentNetWriteBufferSize = netWriteBufferSize();
+            netWriteBuffer.compact();
```

Comment:
Not sure how this work. Will talk offline.

Reply:
I think the bug before happened when some bytes were written/read from the socket but not all.